crates/env/src/chain_extension.rs

```diff
@@ -0,0 +1,403 @@
+    ChainExtensionMethod<I, state::NoResult<O>, state::HandleErrorCode<ErrorCode>>
+where
+    I: scale::Encode,
+    O: scale::Decode,
+    ErrorCode: FromStatusCode,
+{
+    /// Calls the chain extension method for case 1B described [here].
+    ///
+    /// [here]: [`ChainExtensionMethodInstance`]
+    ///
+    /// # Errors
+    ///
+    /// If the called chain extension method returned a non-successful error code.
+    ///
+    /// # Panics
+    ///
+    /// If the returned return value of the called chain extension method cannot be decoded into `O`.
+    ///
+    /// # Example
+    ///
+    /// Declares a chain extension method with the unique ID of 5 that requires a `bool` and an `i32`
+    /// as input parameters and returns a `Result<i32, MyErrorCode>` upon completion.
+    /// It will handle the shared error code from the chain extension.
+    /// The call is finally invoked with arguments `true` and `42` for the `bool` and `i32` input
+    /// parameter respectively.
+    ///
+    /// ```no_compile
+    /// let result = ChainExtensionMethod::build(5)
+    ///     .input::<(bool, i32)>()
+    ///     .output::<i32>()
+    ///     .handle_error_code::<MyErrorCode>()
+    ///     .call(&(true, 42));
+    /// ```
+    #[inline(always)]
+    pub fn call(self, input: &I) -> Result<O, ErrorCode> {
+        <EnvInstance as OnInstance>::on_instance(|instance| {
+            EnvBackend::call_chain_extension::<I, O, ErrorCode, ErrorCode, _, _>(
+                instance,
+                self.func_id,
+                input,
+                ErrorCode::from_status_code,
+                |output| {
+                    let decoded = <O as scale::Decode>::decode(&mut &output[..])
+                        .expect("encountered error while decoding chain extension method call return value");
+                    Ok(decoded)
+                },
+            )
+        })
+    }
+}
+
+impl<I, O> ChainExtensionMethod<I, state::NoResult<O>, state::IgnoreErrorCode>
+where
+    I: scale::Encode,
+    O: scale::Decode,
+{
+    /// Calls the chain extension method for case 2B described [here].
+    ///
+    /// [here]: [`ChainExtensionMethodInstance`]
+    ///
+    /// # Panics
+    ///
+    /// If the returned return value of the called chain extension method cannot be decoded into `O`.
+    ///
+    /// # Example
+    ///
+    /// Declares a chain extension method with the unique ID of 5 that requires a `bool` and an `i32`
+    /// as input parameters and returns a `Result<i32, MyErrorCode>` upon completion.
+    /// It will ignore the shared error code from the chain extension and assumes that the call succeeds.
+    /// The call is finally invoked with arguments `true` and `42` for the `bool` and `i32` input
+    /// parameter respectively.
+    ///
+    /// ```no_compile
+    /// let result = ChainExtensionMethod::build(5)
+    ///     .input::<(bool, i32)>()
+    ///     .output::<i32>()
+    ///     .ignore_error_code::<MyErrorCode>()
```

Comment:
```suggestion
    ///     .ignore_error_code()
```

Reply:
How was this compiling before? :upside_down_face:

Reply:
`no_compile` I guess